tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc

```diff
@@ -2019,6 +2019,118 @@ LogicalResult HandleGpuFusionOp(OpBuilder& b, Operation* fusion) {
   return success();
 }
 
+// For concat op having many operands, we use following schedule:
+//
+// parallel.for %idx in range(num_operands_of_concat) {
+//   if (%idx == 0) {
+//     copy operand #0 to output buffer
+//   } else if (idx == 1) {
+//     copy operand #1 to output buffer
+//   } ...
+//   ... ...
+//   } else {
+//     copy the last operand to output buffer
+//   }
+// }
+
+LogicalResult emitSwitchOperandIdx(OpBuilder& b, Location loc,
```

Comment:
maybe it's better to add some comments with pseudo code to interpret this recursive algorithm.

Reply:
done